src/services/RecordStreamProcessor.ts

```diff
@@ -0,0 +1,110 @@
+  private static instance: RecordStreamProcessor;
+  private readonly logger: Logger;
+  private readonly metrics: Metrics;
+  private readonly validationHelper: ValidationHelper;
+  private readonly iprService: IPRService;
+  private readonly environmentVariables: EnvironmentVariables;
+
+  constructor(logger: Logger, metrics: Metrics) {
+    this.logger = logger;
+    this.environmentVariables = new EnvironmentVariables(logger, ServicesEnum.RECORD_STREAM_PROCESSOR);
+    this.validationHelper = new ValidationHelper();
+    this.metrics = metrics;
+    this.iprService = IPRService.getInstance(this.environmentVariables.sessionEventsTable(), this.logger, createDynamoDbClient());
+  }
+
+  static getInstance(logger: Logger, metrics: Metrics): RecordStreamProcessor {
+    if (!RecordStreamProcessor.instance) {
+      RecordStreamProcessor.instance = new RecordStreamProcessor(logger, metrics);
+    }
+    return RecordStreamProcessor.instance;
+  }
+
+  async processRequest(sessionEvent: any): Promise<void> {
+
+    let sessionEventData: any = ExtSessionEvent.parseRequest(JSON.stringify(sessionEvent));
+
+    this.logger.appendKeys({ govuk_signin_journey_id: sessionEventData.clientSessionId });
+
+    if (sessionEventData.notified) {
+      this.handleNotifiedUser();
+    }
+
+    try {
+      this.validationHelper.validateSessionEventFields(sessionEventData);
+    } catch (error: any) {
+      this.handleValidationFailure(error);
+    }
+
+    let emailType = Constants.VIST_PO_EMAIL_DYNAMIC;
+    if (!sessionEventData.documentUploadedOn || !(sessionEventData.documentUploadedOn > 0)) {
+      const result = this.handleDocumentUploadedOnNotSet(sessionEventData);
+      emailType = result.emailType;
+      sessionEventData = result.sessionEventData;
+    }
+
+    const data = await this.validationHelper.validateSessionEvent(sessionEventData, emailType, this.logger);
+
+    await this.sendEmailMessageToGovNotify(data.sessionEvent, data.emailType);
+
+    try {
+      await this.iprService.setNotifiedFlag(data.sessionEvent.userId, true);
+    } catch (error: any) {
+      throw new AppError(HttpCodesEnum.SERVER_ERROR, error.message);
+    }
+
+  }
+
+  private handleNotifiedUser(): void {
+    this.logger.warn("User is already notified for this session event.", { messageCode: MessageCodes.USER_ALREADY_NOTIFIED });
+    throw new AppError(HttpCodesEnum.SERVER_ERROR, "User is already notified for this session event.");
+  }
+
+  private handleValidationFailure(error: any): void {
+    this.logger.warn(error.message, { messageCode: MessageCodes.MISSING_MANDATORY_FIELDS_IN_SESSION_EVENT });
+    throw new AppError(HttpCodesEnum.SERVER_ERROR, error.message);
+  }
+
+  private handleDocumentUploadedOnNotSet(eventData: any): { emailType: string; sessionEventData: SessionEvent; } {
+    this.logger.info({ message: "documentUploadedOn is not yet populated, sending the old template email." });
+    const emailType = Constants.VIST_PO_EMAIL_STATIC;
```

Comment:
Think this is existing but there is a typo in `VIST_PO_EMAIL_STATIC`

Reply:
Wont the correction of the typo affect the existing events which will have the messageType as VIST_PO_EMAIL_STATIC and may get rejected?

Reply:
It will, I'll just add both cases for now and TODO: remove the wrongly spelt one later